aptos-move/framework/src/natives/aggregator_natives/aggregator_factory.rs

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use aptos_aggregator::aggregator_extension::AggregatorID;
+use aptos_crypto::hash::DefaultHasher;
+use move_deps::{
+    move_binary_format::errors::PartialVMResult,
+    move_core_types::gas_algebra::{InternalGas, InternalGasPerByte, NumBytes},
+    move_vm_runtime::native_functions::{NativeContext, NativeFunction},
+    move_vm_types::{
+        loaded_data::runtime_types::Type,
+        natives::function::NativeResult,
+        pop_arg,
+        values::{Struct, StructRef, Value},
+    },
+};
+use smallvec::smallvec;
+use std::{collections::VecDeque, sync::Arc};
+
+use crate::natives::aggregator_natives::{helpers::get_handle, NativeAggregatorContext};
+
+/***************************************************************************************************
+ * native fun new_aggregator(aggregator_factory: &mut AggregatorFactory, limit: u128): Aggregator;
+ *
+ *   gas cost: base_cost + unit_cost * data_length
+ *
+ **************************************************************************************************/
+#[derive(Debug, Clone)]
+pub struct NewAggregatorGasParameters {
+    pub base_cost: InternalGas,
+    pub unit_cost: InternalGasPerByte,
```

Comment:
I don't think you need a per byte cost here since the size of the data you are dealing is constant. A single base cost should cover everything.